microservices-connector/cmd/router/main_test.go

```diff
@@ -0,0 +1,560 @@
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	mcv1alpha3 "github.com/opea-project/GenAIInfra/genai-microservices-connector/api/v1alpha3"
+	"github.com/stretchr/testify/assert"
+	"knative.dev/pkg/apis"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func init() {
+	logf.SetLogger(zap.New())
+}
+
+func TestSimpleModelChainer(t *testing.T) {
+	// Start a local HTTP server
+	model1 := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		_, err := io.ReadAll(req.Body)
+		if err != nil {
+			return
+		}
+		response := map[string]interface{}{"predictions": "1"}
+		responseBytes, _ := json.Marshal(response)
+		_, err = rw.Write(responseBytes)
+		if err != nil {
+			return
+		}
+	}))
+	model1Url, err := apis.ParseURL(model1.URL)
+	if err != nil {
+		t.Fatalf("Failed to parse model url")
+	}
+	defer model1.Close()
+	model2 := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		_, err := io.ReadAll(req.Body)
+		if err != nil {
+			return
+		}
+		response := map[string]interface{}{"predictions": "2"}
+		responseBytes, _ := json.Marshal(response)
+		_, err = rw.Write(responseBytes)
+		if err != nil {
+			return
+		}
+	}))
+	model2Url, err := apis.ParseURL(model2.URL)
+	if err != nil {
+		t.Fatalf("Failed to parse model url")
+	}
+	defer model2.Close()
+
+	gmcGraph := mcv1alpha3.GMConnector{
+		Spec: mcv1alpha3.GMConnectorSpec{
+			Nodes: map[string]mcv1alpha3.Router{
+				"root": {
+					RouterType: mcv1alpha3.Sequence,
+					Steps: []mcv1alpha3.Step{
+						{
+							StepName:   "model1",
+							ServiceURL: model1Url.String(),
+							Executor: mcv1alpha3.Executor{
+								InternalService: mcv1alpha3.GMCTarget{
+									NameSpace:   "default",
+									ServiceName: "embedding-service",
+								},
+							},
+						},
+						{
+							StepName:   "model2",
+							ServiceURL: model2Url.String(),
+							Executor: mcv1alpha3.Executor{
+								InternalService: mcv1alpha3.GMCTarget{
+									NameSpace:   "default",
+									ServiceName: "tei-embedding-service",
+								},
+							},
+							Data: "$response",
+						},
+					},
+				},
+			},
+		},
+	}
+	input := map[string]interface{}{
+		"instances": []string{
+			"test",
+			"test2",
+		},
+	}
+	jsonBytes, _ := json.Marshal(input)
+	headers := http.Header{
+		"Authorization": {"Bearer Token"},
+	}
+
+	res, _, err := routeStep("root", gmcGraph, jsonBytes, headers)
+	if err != nil {
+		return
+	}
+	var response map[string]interface{}
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return
+	}
+	expectedResponse := map[string]interface{}{
+		"predictions": "2",
+	}
+	fmt.Printf("final response:%v\n", response)
+	assert.Equal(t, expectedResponse, response)
+}
+
+func TestSimpleModelEnsemble(t *testing.T) {
+	// Start a local HTTP server
+	model1 := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
change all 'model' to 'service' : eg: model1 -> service1

Reply:
thanks, done!